polus-feature-extraction-plugin/src/main.py

```diff
@@ -886,196 +890,168 @@ def all(seg_img, units, int_img):
+            if unitLength and not embeddedpixelsize:
+                check_cols = [col for col in df.columns if 'Area' in col]
+                df.columns = [x + '^2' if x in check_cols else x for x in df]
+            df.columns = [c+f'_channel{channel}' for c in df.columns]
+        else:
+            #create dataframe for features selected
+            if label_image is None:
+                df = pd.DataFrame({each_feature: feature_value},index=[0])
+            else:
+                df = pd.DataFrame({each_feature: feature_value})
+            if 'Area' or 'Convex area' or 'Equivalent diameter' or 'Major axis length' or 'Maxferet' or 'Minor axis length' or 'Minferet' or 'Perimeter' in df.columns:
+
+                # Change the units depending on selection
+                if embeddedpixelsize:
+                    units = img_emb_uint
+                elif unitLength and not embeddedpixelsize:
+                    units = unitLength
+                else:
+                    units = "pixels"
+
+                df.rename({
+                    "area": f"Area_{units}",
+                    "convex_area": f"Convex Area_{units}",
+                    "equivalent_diameter": f"Equivalent diameter_{units}", 
+                    "major_axis_length": f"Major axis length_{units}", 
+                    "minor_axis_length": f"Minor axis length_{units}",
+                    "maxferet": f"Maxferet_{units}", 
+                    "minferet": f"Minferet_{units}",
+                    "perimeter": f"Perimeter_{units}"
+                }, axis='columns', inplace=True)
+                columns = [c+f'' for c in df.columns]
+                if unitLength and not embeddedpixelsize:
+                    check_cols = [col for col in df.columns if 'Area' in col]
+                    if check_cols:
+                        df.columns = [col+'^2'for col in check_cols]
+                df.columns = [c+f'_channel{channel}' for c in df.columns]
+        df_insert = pd.concat([df_insert, df], axis=1)
+
+ 
+    if label_image is not None:
+        int_ravel = np.ravel(label_image)
+        int_ravel1 = int_ravel / np.max(int_ravel)
+        int_ravel_bool = np.array_equal(int_ravel1, int_ravel1.astype(bool))
+   
+        #check whether the array contains only zero
+        countzero = not np.any(int_ravel) 
+        if int_ravel_bool == True or countzero == True:
+            df_insert.insert(0, 'Image', title)
+            #Capitalize the first letter of header
+            df_insert.columns = map (lambda x: x.capitalize(), df_insert.columns)
+        else:
+            data = { 'Image':title,
+                    'Label': label, 
+                      'Touching border': border_cells} 
+            df1 = pd.DataFrame(data,columns=['Image','Label','Touching border'])
+            df_insert1 = pd.concat([df1,df_insert],ignore_index=True, axis=1)
+            dfch = df_insert.columns.tolist()
+            df_values= ['Image','Label','Touching border']
```

Comment:
The purpose of this code isn't clear. Why do you need to check to see if it only contains 0? You should add some code comments to explain the purpose of this check. Shouldn't feature extraction just return a `None` value if there are no non-zero labels?

If this is a must, then this code will be slow for large images. Consider using this instead to check if all values are zero:

```python
all_zero = np.all(label_image==0)
```

If you need to check if all values are the same, you can do:
```python
all_equal = np.all(label_image==label_image[0,0])
```